script.js

```diff
@@ -1,12 +1,210 @@
+    option.textContent = show.name;
+    selectShows.appendChild(option);
+  });
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+// Search specific Shows based on select
+async function selectShow(event) {
+  searchEpisodes.value = "";
+  if (event.target.value == "allSerials") {
+    makeCardForShows(sortedShows);
+    //Remove options from episodes's select
+    const options = selectEpisodes.getElementsByTagName("option");
+    for (let i = options.length - 1; i > 0; i--) {
+      selectEpisodes.removeChild(options[i]);
+    }
+    searchEpisodes.disabled = true;
+  } else {
+    searchEpisodes.disabled = false;
+    //Remove options from episodes's select
+    const options = selectEpisodes.getElementsByTagName("option");
+    for (let i = options.length - 1; i > 0; i--) {
+      selectEpisodes.removeChild(options[i]);
+    }
+    // Find selected movie
+    const specificShow = sortedShows.find(
+      (show) => show.name === event.target.value
+    );
+    // Determine show's id
+    const specificShowID = specificShow.id;
+    // Get all show's episodes
+    allEpisodes = await getAllAvailableEpisodes(specificShowID);
+    // make card for each episode
+    makeCardForEpisodes(allEpisodes);
+  }
 }
 
-window.onload = setup;
+//Get all episodes via fetch
+async function getAllAvailableEpisodes(showId) {
+  // await response of the fetch call
+  const response = await fetch(
+    `https://api.tvmaze.com/shows/${showId}/episodes`
+  );
+  // only proceed once its resolved
+  const data = await response.json();
+  // only proceed once second promise is resolved
+  return data;
+}
+
+//make card for each episode
+function makeCardForEpisodes(episodeList) {
+  let output = "";
+  episodeList.forEach((episode) => {
+    const paddedEpisodeSeason = episode.season.toString().padStart(2, "0");
+    const paddedEpisodeNumber = episode.number.toString().padStart(2, "0");
+    const episodeIdentifier = `S${paddedEpisodeSeason}E${paddedEpisodeNumber}`;
+    output += `
+      <div class="col">
+        <div class="card h-100 px-0 mx-1">
+        <img src="${episode.image.medium}" class="card-img-top mx-0" alt="...">
+        <div class="card-body mb-4">
+        <h5 >${episode.name} - ${episodeIdentifier}</h5>
+      <p>${episode.summary}</p>
+        </div>
+        <a href="${episode.url}" class="btn btn-primary" style="position:absolute; bottom:10px; left:20px; right:20px;" target="_blank">See More</a>
+      </div>
+      </div>
+    `;
+  });
+  showEpisodseRow.innerHTML = output;
+
+  // Create option for each episodes in select after loading page
+  episodeList.forEach((episode) => {
+    const option = document.createElement("option");
+    option.value = `S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")}`;
+    option.textContent = `S${episode.season
```

Comment:
This is the third place I've seen you write the code to turn `{season: 2, number: 6}` into `"S02E06"` - can you extract a function to do this? Why is it a good idea to extract a function?